src/Database/Builder.php

```diff
@@ -128,12 +128,12 @@ protected function searchWhereInternal($term, $columns, $mode, $boolean)
      * @param string|int|null $pageName
      * @return \Illuminate\Contracts\Pagination\LengthAwarePaginator
      */
-    public function paginate($perPage = null, $currentPage = null, $columns = ['*'], $pageName = 'page')
+    public function paginate($perPage = null, $currentPage = null, $columns = ['*'], $pageName = 'page', $total = 0)
```

Comment:
This has been added to the signature but it doesn't look like it's being used, see line 156

Reply:
I added code to take it into account if provided.